packages/neon-domain/src/provider/common.ts

```diff
@@ -0,0 +1,27 @@
+import { logging, rpc, sc, u } from "@cityofzion/neon-core";
+
+export interface DomainProvider {
+  name: string;
+  resolveDomain(url: string, domain: string, tld: string): Promise<string>;

```

Comment:
is it possible for us to simplify this down to just `(url, domain)`? From a user perspective, I think many are used to just typing in the entire address as opposed to having separate fields. We can do the check for tld internally. 

This also seems like an implementation detail that is provider specific.